src/bar.js

```diff
@@ -186,15 +190,64 @@ export default class Bar {
             }
 
             this.gantt.unselect_all();
-            this.group.classList.toggle('active');
+            
+            //  add dependency by popup button
+            if(this.gantt.dependencyBar != null){
+            	this.add_dependency();
+            }else{
+                this.group.classList.toggle('active');
+                this.show_popup();
+            }
 
-            this.show_popup();
         });
     }
 
+    add_dependency(){
+		// already marked a dependency
+		var markedTask = this.gantt.dependencyBar.task;
+		if(markedTask == null)
+			return;
```

Comment:
```suggestion
		if (selected_task === null) {
			return;
		}
```

Always wrap single line `if` statements in curly braces. Please do this everywhere you have done your changes.